modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -747,12 +763,180 @@ public void testUpdateGroupWithPUTNotImplementedException(String id, String scim
+            throws BadRequestException, CharonException, NotImplementedException, NotFoundException {
+
+        Group groupNew = (Group) scimNewGroupObject;
+        Group groupOld = (Group) scimOldGroupObject;
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        Mockito.when(userManager.getGroup(anyString(), anyMap())).thenReturn(groupOld);
+        Mockito.when(userManager.patchGroup(anyString(), anyString(), anyMap(), anyMap())).thenReturn(groupNew);
+        SCIMResponse scimResponse = groupResourceManager.updateWithPATCH(existingId, patchRequest,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHOverride")
+    public Object[][] dataToUpdateWithPATCHOverride()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new Group());
+        return new Object[][]{
+                {id, SCIM2_PATCH_REQUEST_STRING, groupNew, groupNew}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCHOverride")
+    public void testUpdateWithPATCHOverride(String existingId, String patchRequest,
+                                            Object scimNewGroupObject, Object scimOldGroupObject)
+            throws BadRequestException, CharonException, NotImplementedException, NotFoundException {
+
+        Group groupNew = (Group) scimNewGroupObject;
+        Group groupOld = (Group) scimOldGroupObject;
+
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        Mockito.when(userManager.getGroup(anyString(), anyMap())).thenReturn(groupOld);
+        Mockito.when(userManager.patchGroup(anyString(), anyString(), anyMap(), anyMap())).thenReturn(groupNew);
+        SCIMResponse scimResponse = groupResourceManager.updateWithPATCH(existingId, patchRequest,
+                userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NO_CONTENT);
+    }
+
+    @DataProvider(name = "dataForListWithGET")
+    public Object[][] dataToListWithGET() {
+
+        return new Object[][]{
+                {null, 1, 2, null, null, "PRIMARY", "members", null},
+                {"displayName sw PRIMARY/manager", 1, 2, null, null, "PRIMARY", "members",
```

Comment:
shall we have data sets to capture all the filters defined in https://github.com/wso2/charon/blob/master/modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMConstants.java#L723-L732 ?

Reply:
is this fixed?

Reply:
Can we shift that to another open PR ?

Reply:
Added in commit c0eb0357fe4b40cf920a29e78da8d87bc75bd86f